### PR TITLE
Get deployment's stage log from deployment-id

### LINKED ISSUE #4159: Get deployment's stage logs from pipectl
**What would you like to be added**:

**Why is this needed**:

### PR DESCRIPTION
**Pull request by funera1**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #4159

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by funera1** [comment]
Change to get stage log from deployment-id only

**Comment by knanao** [comment]
@funera1 
Do you need to fix this?
This seems that you're still in the middle of fixing it.
If so, please make this WIP temporarily.

**Comment by funera1** [comment]
Sorry. I will make this WIP temporarily.

File: pkg/app/server/service/apiservice/service.proto [hunk 1]
```diff
@@ -195,3 +198,17 @@ message EncryptRequest {
 message EncryptResponse {
     string ciphertext = 1 [(validate.rules).string.min_len = 1];
 }
+
+message StageLog {
+    repeated model.LogBlock blocks = 1;
+    bool completed = 2;
+    string message = 3;
```
Comment (funera1) [inline pkg/app/server/service/apiservice/service.proto:205]: Remove

File: pkg/app/pipectl/cmd/deployment/logs.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+func newLogsCommand(root *command) *cobra.Command {
+	c := &logs{
+		root:   root,
+		stdout: os.Stdout,
+	}
+	cmd := &cobra.Command{
+		Use:   "logs",
+		Short: "Get deployment stage logs.",
```
Comment (knanao) [inline pkg/app/pipectl/cmd/deployment/logs.go:45]: If this is just for getting the stage log of deployments, this subcommand name is too luxurious.
How about adding this command to the deployment sub-command?

File: pkg/app/server/grpcapi/api.go [hunk 3]
```diff
@@ -375,6 +381,49 @@ func (a *API) GetDeployment(ctx context.Context, req *apiservice.GetDeploymentRe
...
+		if err != nil && !errors.Is(err, stagelogstore.ErrNotFound) {
+			return nil, err
+		}
+
+		if err != nil && stage.Name == model.StageRollback.String() {
+			// Rollback is generated automatically and returns nothing if not found
+			if stage.Name == model.StageRollback.String() {
+				continue
+			} else {
+				stageLogs[stage.Id] = &apiservice.StageLog{}
```
Comment (knanao) [inline pkg/app/server/grpcapi/api.go:412]: This case will not be called never.

File: pkg/app/pipectl/cmd/deployment/logs.go (hunk 2 above)
Comment (funera1) [reply pkg/app/pipectl/cmd/deployment/logs.go:45]: Hmmm
How about pipectl deployments logs stage-log?

Comment (knanao) [reply pkg/app/pipectl/cmd/deployment/logs.go:45]: I feel `pipectl deployment logs` is fine. wdyt?

File: pkg/app/server/grpcapi/api.go [hunk 4]
```diff
@@ -375,6 +381,49 @@ func (a *API) GetDeployment(ctx context.Context, req *apiservice.GetDeploymentRe
...
+		}
+
+		if err != nil {
+			// Rollback is generated automatically and returns nothing if not found
+			if stage.Name == model.StageRollback.String() {
+				continue
+			} else {
+				stageLogs[stage.Id] = &apiservice.StageLog{}
+			}
+		}
```
Comment (knanao) [inline pkg/app/server/grpcapi/api.go:414]:
```suggestion
		// StageRollback is generated automatically and returns nothing if not found.
		if err != nil  && stage.Name == model.StageRollback.String() {
			continue
		}
		if err != nil {
			stageLogs[stage.Id] = &apiservice.StageLog{}
		}
```

**Comment by knanao** [comment]
@funera1 
Please run `make gen/code` and push the generated file.

**Review by knanao — APPROVED** [review]
Nice work.

**Pull request merged**